proxy_list.go

```diff
@@ -0,0 +1,332 @@
+func (l *List) AddAll(elements ...interface{}) (bool, error) {
+	elementsData, err := l.validateAndSerializeValues(elements...)
+	if err != nil {
+		return false, err
+	}
+	request := codec.EncodeListAddAllRequest(l.name, elementsData)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return false, err
+	}
+	return codec.DecodeListAddAllResponse(response), nil
+}
+
+// AddAllAt inserts all elements in the specified slice at specified index, keeping the order of the slice.
+// Shifts the subsequent elements to the right.
+// Returns true if the list has changed as a result of this operation, false otherwise.
+func (l *List) AddAllAt(index int32, elements ...interface{}) (bool, error) {
+	elementsData, err := l.validateAndSerializeValues(elements...)
+	if err != nil {
+		return false, err
+	}
+	request := codec.EncodeListAddAllWithIndexRequest(l.name, index, elementsData)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return false, err
+	}
+	return codec.DecodeListAddAllWithIndexResponse(response), nil
+}
+
+// Clear removes all elements from the list.
+func (l *List) Clear() error {
+	request := codec.EncodeListClearRequest(l.name)
+	_, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	return err
+}
+
+// Contains checks if the list contains the given element.
+// Returns true if the list contains the element, false otherwise.
+func (l *List) Contains(element interface{}) (bool, error) {
+	elementData, err := l.validateAndSerialize(element)
+	if err != nil {
+		return false, err
+	}
+	request := codec.EncodeListContainsRequest(l.name, elementData)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return false, err
+	}
+	return codec.DecodeListContainsResponse(response), nil
+}
+
+// ContainsAll checks if the list contains all of the given elements.
+// Returns true if the list contains all of the elements, otherwise false.
+func (l *List) ContainsAll(elements ...interface{}) (bool, error) {
+	elementsData, err := l.validateAndSerializeValues(elements...)
+	if err != nil {
+		return false, err
+	}
+	request := codec.EncodeListContainsAllRequest(l.name, elementsData)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return false, err
+	}
+	return codec.DecodeListContainsAllResponse(response), nil
+}
+
+// Get retrieves the element at given index.
+func (l *List) Get(index int32) (interface{}, error) {
+	request := codec.EncodeListGetRequest(l.name, index)
+	response, err := l.invokeOnPartition(context.TODO(), request, l.partitionID)
+	if err != nil {
+		return nil, err
+	}
+	return l.convertToObject(codec.DecodeListGetResponse(response))
+}
+
+// IndexOf returns the index of the first occurrence of the given element in this list.
+func (l *List) IndexOf(element interface{}) (int32, error) {
```

Comment:
The same comment about `int32` vs `int` as the `index`...

Reply:
For the returned indexes we may use `int`, but I'm not sure if it's worth it. Having the same type for in and out values has the should be more obvious for the user due to the symmetry.